lib/auth/register.go

```diff
@@ -740,23 +756,89 @@ func registerUsingAzureMethod(
 		}
 
 		return &proto.RegisterUsingAzureMethodRequest{
-			RegisterUsingTokenRequest: &types.RegisterUsingTokenRequest{
-				Token:                token,
-				HostID:               params.ID.HostUUID,
-				NodeName:             params.ID.NodeName,
-				Role:                 params.ID.Role,
-				AdditionalPrincipals: params.AdditionalPrincipals,
-				DNSNames:             params.DNSNames,
-				PublicTLSKey:         params.PublicTLSKey,
-				PublicSSHKey:         params.PublicSSHKey,
-			},
-			AttestedData: ad,
-			AccessToken:  accessToken,
+			RegisterUsingTokenRequest: registerUsingTokenRequestForParams(token, params),
+			AttestedData:              ad,
+			AccessToken:               accessToken,
 		}, nil
 	})
 	return certs, trace.Wrap(err)
 }
 
+// registerUsingTPMMethod is used to register using the TPM join method. It
+// is able to register through a proxy or through the auth server directly.
+func registerUsingTPMMethod(
```

Comment:
Is this client-side code? Why is it under lib/auth?

Reply:
One might ask the same question of the entirety of `auth/register.go` 😓 I'm pursuing a longer term goal of breaking `tbot`'s dependency on `lib/auth`, and in the course of that, `register.go` should eventually find itself under a `joining` or similar package.

Reply:
OK, thanks for the context.